WordPress/Classes/Models/BlogSettings.swift

```diff
@@ -0,0 +1,203 @@
+     */
+    @NSManaged var commentsRequireNameAndEmail : NSNumber?
+    
+    /**
+     *  @details Specifies whether commenters should be registered or not.
+     */
+    @NSManaged var commentsRequireRegistration : NSNumber?
+    
+    /**
+     *  @details Indicates the sorting order of the comments. Ascending / Descending, based on the date.
+     */
+    @NSManaged var commentsSortOrder : NSNumber?
+    
+    /**
+     *  @details Indicates the number of levels allowed per comment.
+     */
+    @NSManaged var commentsThreadingDepth : NSNumber?
+    
+    /**
+     *  @details When enabled, comment threading will be supported.
+     */
+    @NSManaged var commentsThreadingEnabled : NSNumber?
+
+    /**
+     *  @details *LOCAL* flag (non stored remotely) indicating whether post geolocation is enabled or not.
+     *           This can be overriden on a per-post basis.
+     */
+    @NSManaged var geolocationEnabled : NSNumber?
+    
+    /**
+     *  @details If set to true, 3rd party sites will be allowed to post pingbacks.
+     */
+    @NSManaged var pingbackInboundEnabled : NSNumber?
+    
+    /**
+     *  @details When Outbound Pingbacks are enabled, 3rd party sites that get linked will be notified.
+     */
+    @NSManaged var pingbackOutboundEnabled : NSNumber?
+
+    
+    
+    // MARK: - Related Posts
+    
+    /**
+    *  @details When set to true, Related Posts will be allowed.
+    */
+    @NSManaged var relatedPostsAllowed : NSNumber?
+    
+    /**
+    *  @details When set to true, Related Posts will be enabled.
+     */
+    @NSManaged var relatedPostsEnabled : NSNumber?
+    
+    /**
+     *  @details Indicates whether related posts should show a headline.
+     */
+    @NSManaged var relatedPostsShowHeadline : NSNumber?
+    
+    /**
+     *  @details Indicates whether related posts should show thumbnails.
+     */
+    @NSManaged var relatedPostsShowThumbnails : NSNumber?
+    
+    
+    
+    
+    /**
+     *  @details Computed property, meant to help conversion from Remote / String-Based values, into their
+     *           Integer counterparts
+     */
+    var commentsSortOrderAsString : String? {
+        set {
+            let isAscending = (newValue == SortOrder.AscendingStringValue)
+            commentsSortOrder = isAscending ? SortOrder.Ascending.rawValue : SortOrder.Descending.rawValue
+        }
+        get {
+            let isAscending = (commentsSortOrder == SortOrder.Ascending.rawValue)
+            return isAscending ? SortOrder.AscendingStringValue : SortOrder.DescendingStringValue
+        }
+    }
+    
+    /**
+     *  @enum    BlogSettings.SortOrder
+     *  @brief   Enumerates all of the valid Sorting Orders
+     */
+    public enum SortOrder : Int {
+        case Ascending  = 0
+        case Descending = 1
+        
+        static let AscendingStringValue  = "asc"
+        static let DescendingStringValue = "desc"
+    }
```

Comment:
I was going to suggest using `SortOrder` as the type for `commentsSortOrder`, but since it has to be used from Obj-C we'd still have to map from/to strings. You could store it as a string, and it would help a bit with conversion, but I'm not sure if this approach is really better

https://gist.github.com/koke/67ac779f7b259728b98e

Although a question came up while playing with this. Does it make sense to allow nil? I understand there's the case where the setting has some value on the server but we don't have it yet. But then `commentsSortOrderAsString` will return `desc` instead of `nil`